src/webgpu/api/validation/getBindGroupLayout.spec.ts

```diff
@@ -0,0 +1,70 @@
+export const description = `
+  getBindGroupLayout validation tests.
+`;
+
+import { makeTestGroup } from '../../../common/framework/test_group.js';
+
+import { ValidationTest } from './validation_test.js';
+
+export const g = makeTestGroup(ValidationTest);
+
+g.test('index_range')
+  .desc(
+    `
+  Test that a validation error is generated if the index exceeds the size of the bind group layouts.
+  `
+  )
+  .params(u => u.combine('index', [0, 1, 2, 3, 4, 5]))
+  .fn(async t => {
+    const { index } = t.params;
+
+    const maxBindGroups = t.device.limits?.maxBindGroups ?? 4;
+
+    const pipelineBufferBindings: GPUBindGroupLayoutEntry[] = [];
+    for (let binding = 0; binding < maxBindGroups; binding++) {
+      pipelineBufferBindings.push({
+        binding,
+        visibility: GPUShaderStage.COMPUTE,
+        buffer: { type: 'uniform' },
+      });
+    }
+
+    const pipelineBindGroupLayouts = t.device.createBindGroupLayout({
+      entries: pipelineBufferBindings,
+    });
+
+    const pipelineLayout = t.device.createPipelineLayout({
+      bindGroupLayouts: [pipelineBindGroupLayouts],
+    });
+
+    const pipeline = t.device.createRenderPipeline({
+      layout: pipelineLayout,
+      vertex: {
+        module: t.device.createShaderModule({
+          code: `
+            @vertex
+            fn main()-> @builtin(position) vec4<f32> {
+              return vec4<f32>(0.0, 0.0, 0.0, 1.0);
+            }`,
+        }),
+        entryPoint: 'main',
+      },
+      fragment: {
+        module: t.device.createShaderModule({
+          code: `
+            @fragment
+            fn main() -> @location(0) vec4<f32> {
+              return vec4<f32>(0.0, 1.0, 0.0, 1.0);
+            }`,
+        }),
+        entryPoint: 'main',
+        targets: [{ format: 'rgba8unorm' }],
+      },
+    });
+
+    const shouldError = index >= maxBindGroups;
+
+    t.expectValidationError(() => {
+      pipeline.getBindGroupLayout(index);
```

Comment:
According to the spec, at least as currently written, this is supposed to throw an exception if `index >= maxBindGroups`. It's only a validation error if it's `< maxBindGroups` but `>=` the number of bind groups actually in the layout.

HOWEVER: I actually think we should change this in the spec. It doesn't seem consistent with other parts of the API. I'm going to open a spec proposal about it. Let's see if that gets resolved quickly.

Note, there's also an issue where `"auto"` layouts always fill the array up to `maxBindGroups`, instead of only up to the highest bind group index needed. I think this is a bug in the spec, so ignore this for now.

Reply:
https://github.com/gpuweb/gpuweb/pull/3505

https://github.com/gpuweb/gpuweb/pull/3506

Hoping both can be resolved tomorrow.

Reply:
Thanks for checking it. Once let me update this PR with some fixes you pointed out.

Reply:
There have been no objections so far, so let's move forward assuming they'll land.